src/main/java/com/fontys/api/service/TournamentService.java

```diff
@@ -80,11 +86,41 @@ public String deleteTournament(Integer id) {
         return Optional.empty();
     }
 
-    private void validateTournament(Integer id) throws InvalidAttributeValueException {
-        if (tournamentRepository.findById(id).isEmpty()) {
+    @Transactional
+    public String addMatchToTournament(Integer tournamentId, Integer matchId)
+    {
+        Optional<Tournament> tournament = tournamentRepository.findById(tournamentId);
+        Optional<Match> match = matchRepository.findById(matchId);
+
+        if(tournament.isEmpty())
+        {
+            return "The tournament does not exist";
+        }
+        if(match.isEmpty())
+        {
+            return "The match does not exist";
+        }
+        Tournament tournament1 = tournament.get();
+        Match match1 = match.get();
+
+        if(tournament1.getMatches().contains(match1))
+        {
```

Comment:
ditto

Reply:
The code has been changed according to your feedback.

Reply:
Resolving this becuase there in a newer version